### PR TITLE
Thermostat Cluster Init Function Enabled

### PR DESCRIPTION
**Pull request by chrisbeach59**
Problem:
emberAfThermostatClusterServerInitCallback was not enabled in the helper.js file for code gen. 
FeatureMap usage involves Conditional compiles and repeated code

Changes:
Enabled Init for thermostat
Eliminated #ifndefs
Used Feature Map accessor function now available in SDK, in place of deprecated read.

Tested:
Manual testing with thermostat-app (Linux) and chip-tool

### CODE REVIEW

File: src/app/clusters/thermostat-server/thermostat-server.cpp [hunk 1]
```diff
@@ -39,6 +39,20 @@ using namespace chip;
...
 
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace Thermostat {
+static uint32_t sFeatureMap;
+bool AutoSupported      = false;
+bool HeatSupported      = false;
+bool CoolSupported      = false;
+bool OccupancySupported = false;
```
Comment (bzbarsky-apple) [inline src/app/clusters/thermostat-server/thermostat-server.cpp:50]: Is there a reason these are not static?

But also, this is all assuming the thermostat cluster is only present on one endpoint, right?

Comment (chrisbeach59) [reply src/app/clusters/thermostat-server/thermostat-server.cpp:50]: Cut and paste for not being static.

You do raise and interesting point about the assumption that there is only one thermostat 
endpoint. 

I will move the feature map reads back into the functions that require the information.

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
Fast tracking given this has been in review for > 3 days.

File: src/app/clusters/thermostat-server/thermostat-server.cpp [hunk 2]
```diff
@@ -105,43 +104,40 @@ MatterThermostatClusterServerPreAttributeChangedCallback(const app::ConcreteAttr
...
-                               reinterpret_cast<uint8_t *>(&FeatureMap), sizeof(FeatureMap));
-#else
-    FeatureMap = FEATURE_MAP_OVERIDE;
-#endif
+    if (FeatureMap::Get(endpoint, &OurFeatureMap) != EMBER_ZCL_STATUS_SUCCESS)
+        OurFeatureMap = FEATURE_MAP_DEFAULT;
 
-    if (FeatureMap & 1 << 5) // Bit 5 is Auto Mode supported
-    {
+    if (OurFeatureMap & 1 << 5) // Bit 5 is Auto Mode supported
```
Comment (andy31415) [inline src/app/clusters/thermostat-server/thermostat-server.cpp:122]: we should have constants for these. Would be much easier to rid:

```cpp
constexpr uint16_t kFeatureMap_AutoModeBit = 1 << 5;
// ....

bool AutoSupported = (OurFeatureMap & kFeatureMap_AutoModeBit) != 0;
```

File: src/app/clusters/thermostat-server/thermostat-server.cpp [hunk 3]
```diff
@@ -105,43 +104,40 @@ MatterThermostatClusterServerPreAttributeChangedCallback(const app::ConcreteAttr
...
+    if (OurFeatureMap & 1 << 0)
         HeatSupported = true;
 
-    if (FeatureMap & 1 << 1)
+    if (OurFeatureMap & 1 << 1)
         CoolSupported = true;
 
-    if (FeatureMap & 1 << 2)
-        OccupancySupported = true;
+    if (OurFeatureMap & 1 << 2)
```
Comment (andy31415) [inline src/app/clusters/thermostat-server/thermostat-server.cpp:131]: Previous code I could read "OccupancySupported = true;" to figure out what this bit 2 is.

new code is totally opaque. At least a comment should be here or ideally constants that are readable.

**Pull request merged**